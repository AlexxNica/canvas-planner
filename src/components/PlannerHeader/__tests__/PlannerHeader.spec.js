import React from 'react';
import { shallow, mount } from 'enzyme';
import { PlannerHeader } from '../index';

const defaultProps = {
  courses: [],
  savePlannerItem: () => {},
  locale: 'en',
  timeZone: 'America/Denver',
  deletePlannerItem: () => {},
};

it('renders the base component correctly with two buttons and a tray', () => {
  const wrapper = shallow(
    <PlannerHeader {...defaultProps} />
  );
  expect(wrapper).toMatchSnapshot();
});

it('toggles the new item tray', () => {
  const wrapper = mount(
    <PlannerHeader {...defaultProps} />
  );
  const button = wrapper.find('[children="Add Note to Self"]');
  button.simulate('click');
  expect(wrapper.find('Tray').props().isOpen).toEqual(true);
  button.simulate('click');
  expect(wrapper.find('Tray').props().isOpen).toEqual(false);
});

it('sends focus back to the add new item button', () => {
  const wrapper = mount(
    <PlannerHeader {...defaultProps} />
  );
  wrapper.setState({trayOpen: true});
  const btn = wrapper.instance().addNoteBtn;
  wrapper.instance().noteBtnOnClose();
  expect(btn.focused).toBe(true);
});
