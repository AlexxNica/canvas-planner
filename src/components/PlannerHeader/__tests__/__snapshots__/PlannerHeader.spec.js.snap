// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the base component correctly with two buttons and a tray 1`] = `
<div>
  <Button
    as="button"
    buttonRef={[Function]}
    fluidWidth={false}
    margin="0"
    onClick={[Function]}
    size="medium"
    type="button"
    variant="icon"
  >
    <IconPlusLine
      title="Add Note to Self"
    />
  </Button>
  <Button
    as="button"
    buttonRef={[Function]}
    fluidWidth={false}
    margin="0"
    size="medium"
    type="button"
    variant="icon"
  >
    <Badge
      count={4}
      formatOverflowText={[Function]}
      placement="top end"
      pulse={false}
      standalone={false}
      type="count"
      variant="primary"
    >
      <IconAlertLine
        title="4 opportunities"
      />
    </Badge>
  </Button>
  <Tray
    border={false}
    closeButtonLabel="Close"
    closeButtonRef={[Function]}
    closeButtonVariant="icon"
    contentRef={[Function]}
    isDismissable={true}
    isOpen={false}
    label="Create/update planner item"
    onExited={[Function]}
    onExiting={[Function]}
    onRequestClose={[Function]}
    placement="end"
    shadow={true}
    trapFocus={true}
  >
    <UpdateItemTray
      courses={Array []}
      locale="en"
      onDeletePlannerItem={[Function]}
      onSavePlannerItem={[Function]}
      timeZone="America/Denver"
    />
  </Tray>
</div>
`;
